app/com/gu/floodgate/DynamoDBTable.scala

```diff
@@ -71,6 +84,20 @@ trait DynamoDBTable[T] extends StrictLogging {
     getListOfItems(request)
   }
 
+  def getLatestItem(id: String, filterKeyName: String, filterValue: String)(implicit ec: ExecutionContext): Future[Option[T]] = {
+    val request = new QueryRequest().withTableName(tableName)
+      .withScanIndexForward(false)
+      .withKeyConditionExpression(s"$keyName = :$keyName")
```

Comment:
the second subst isn't really needed, is it? i.e. `:keyName` is enough, and then `Map(":keyName" -> ...)`